packages/autorest.typescript/src/transforms/extensions.ts

```diff
@@ -204,9 +204,22 @@ function addPageableMethods(codeModel: CodeModel) {
         }
       );
 
+      // Filter out query parameters from the original operation.
+      if (nextLinkMethod.parameters) {
+        nextLinkMethod.parameters = nextLinkMethod.parameters.filter(param => {
+          return param.protocol.http?.in !== ParameterLocation.Query;
+        });
```

Comment:
Also, could this change be a breaking change? are there query params out there that are expected to be added to the nextLink URL for some reason? I am not saying this is an ideal behavior, I just wonder if we're breaking any one.

Reply:
While I'd never say never because we've seen some pretty wild service behaviors, at least according to our API guidelines this shouldn't be a problem: https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#981-server-driven-paging

> Clients MUST treat the continuation URL as opaque, which means that query options may not be changed while iterating over a set of partial results.